plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/impls/redshift/RedshiftView.java

```diff
@@ -98,43 +99,34 @@ private boolean isViewVwithNoSchemaBinding(DBRProgressMonitor monitor) throws DB
                         String colType = JDBCUtils.safeGetString(dbResult, "col_type");
                         int colNum = JDBCUtils.safeGetInt(dbResult, "col_num");
 
-                        PostgreTableColumn viewColumn = new PostgreTableColumn(this);
-                        viewColumn.setName(colName);
-                        viewColumn.setPersisted(true);
-
-                        String colTypeName;
-                        int colPrecision = -1, colScale = -1;
-                        int divPos = colType.indexOf("(");
-                        if (divPos != -1) {
-                            colTypeName = colType.substring(0, divPos);
-                            int divPos2 = colType.lastIndexOf(")");
-                            if (divPos2 < 0) {
-                                log.error("Bad column type name '" + colType + "'");
-                                continue;
-                            }
-                            String typeMod = colType.substring(divPos + 1, divPos2).trim();
-                            divPos2 = typeMod.indexOf(",");
-                            if (divPos2 < 0) {
-                                colPrecision = CommonUtils.toInt(typeMod);
-                            } else {
-                                colPrecision = CommonUtils.toInt(typeMod.substring(0, divPos2).trim());
-                                colScale = CommonUtils.toInt(typeMod.substring(divPos2 + 1).trim());
-                            }
-                        } else {
-                            colTypeName = colType;
-                        }
+			String colTypeName;
+			int divPos = colType.indexOf("(");
+			if (divPos != -1) {
+			    colTypeName = colType.substring(0, divPos);
+			    int divPos2 = colType.lastIndexOf(")");
+			    if (divPos2 < 0) {
+				log.error("Bad column type name '" + colType + "'");
+				continue;
+			    }
+			} else {
+			    colTypeName = colType;
+			}
+                        
+			String resolvedColTypeName = PostgreConstants.DATA_TYPE_ALIASES.get(colTypeName);
+			colTypeName = resolvedColTypeName == null ? colTypeName : resolvedColTypeName;
 
                         PostgreDataType dataType = (PostgreDataType) getDataSource().resolveDataType(monitor, colTypeName);
                         if (dataType == null) {
                             log.error("Column type name '" + colType + "' not found");
                             continue;
                         }
+                        PostgreTableColumn viewColumn = new PostgreTableColumn(this);
+                        viewColumn.setName(colName);
+                        viewColumn.setFullTypeName(colType);
+                        viewColumn.setPersisted(true);
                         viewColumn.setDataType(dataType);
-                        if (colPrecision >= 0) viewColumn.setPrecision(colPrecision);
-                        if (colScale >= 0) viewColumn.setScale(colScale);
```

Comment:
I guess complex type handlers are not working anymore, so you can't, for example, edit string values.

You can at least use `PostgreTypeHandlerProvider#getTypeHandler` here and then `setTypeMod` on `viewColumn` using `PostgreTypeHandler#getTypeModifiers` using your newly acquired type handler. No need to update precision and scale after that though

Reply:
@ShadelessFox 
I can make this change. Can you pls elaborate on what you mean by edit of string values don’t work?

Reply:
Typemod was not available in the late binding view. That’s the reason I went with this approach. I do see numeric is displayed right (see screenshot  above in my first comment). 

Redshift already gives the data in a displayable format. 

The output from Redshift itself is numeric(18, 7).

While they retained typemod approach for tables and normal views, for late binding views all they provide is col_type. 

In hindsight, Why complicate it with type modifier, when current approach is working.

Reply:
I see, okay. I forgot that `PostgreAttribute#setFullTypeName` already does that I just described to you.

Reply:
Thanks @ShadelessFox. I completed the other change you suggested. Hope we are all good with this PR.